lib/install.ps1

```diff
@@ -941,12 +963,61 @@ function show_suggestions($suggested) {
     }
 }
 
-# travelling directories have their contents moved from
-# $from to $to when the app is updated.
-# any files or directories that already exist in $to are skipped
-function travel_dir($from, $to) {
-    $skip_dirs = ls $to -dir | % { "`"$from\$_`"" }
-    $skip_files = ls $to -file | % { "`"$from\$_`"" }
+# Persistent data
+function persist_def($persist) {
+    if ($persist -is [Array]) {
+        $source = $persist[0]
+        $target = $persist[1]
+    } else {
+        $source = $persist
+        $target = $null
+    }
 
-    robocopy $from $to /s /move /xd $skip_dirs /xf $skip_files > $null
+    if (!$target) {
+        $target = fname($source)
+    }
+
+    return $source, $target
+}
+
+function persist_data($manifest) {
+    $persist = $manifest.persist
+    if($persist) {
+        $persist_dir = ensure $persist_dir
```

Comment:
As above for passing this variable explicitly rather than inheriting it from parent scope.